ui/lib/apps/SlowQuery/components/List.tsx

```diff
@@ -0,0 +1,209 @@
+} from './TimeRangeSelector'
+import { useTranslation } from 'react-i18next'
+import client, { SlowqueryBase } from '@lib/client'
+import { ReloadOutlined } from '@ant-design/icons'
+import { IColumn } from 'office-ui-fabric-react/lib/DetailsList'
+import * as useSlowQueryColumn from '../utils/useColumn'
+import DetailPage from './Detail'
+
+const { Option } = Select
+const { Search } = Input
+
+const tableColumns = (
+  rows: SlowqueryBase[],
+  onColumnClick: (ev: React.MouseEvent<HTMLElement>, column: IColumn) => void,
+  orderBy: string,
+  desc: boolean,
+  showFullSQL?: boolean
+): IColumn[] => {
+  return [
+    useSlowQueryColumn.useInstanceColumn(rows),
+    useSlowQueryColumn.useConnectionIDColumn(rows),
+    useSlowQueryColumn.useDigestColumn(rows, showFullSQL),
+    useSlowQueryColumn.useEndTimeColumn(rows),
+    {
+      ...useSlowQueryColumn.useQueryTimeColumn(rows),
+      isSorted: orderBy === 'Query_time',
+      isSortedDescending: desc,
+      onColumnClick: onColumnClick,
+    },
+    {
+      ...useSlowQueryColumn.useMemoryColumn(rows),
+      isSorted: orderBy === 'Mem_max',
+      isSortedDescending: desc,
+      onColumnClick: onColumnClick,
+    },
+  ]
+}
+
+export default function List() {
+  const navigate = useNavigate()
+  const { t } = useTranslation()
+
+  const [curTimeRange, setCurTimeRange] = useState<TimeRange>(DEF_TIME_RANGE)
+  const [curSchemas, setCurSchemas] = useState<string[]>([])
+  const [schemas, setSchemas] = useState<string[]>([])
+  const [searchText, setSearchText] = useState('')
+  const [orderBy, setOrderBy] = useState('Query_time')
+  const [desc, setDesc] = useState(true)
+  const [limit, setLimit] = useState(50)
+  const [refreshTimes, setRefreshTimes] = useState(0)
+
+  const [loading, setLoading] = useState(false)
+  const [slowQueryList, setSlowQueryList] = useState<SlowqueryBase[]>([])
+
+  const [columns, setColumns] = useState<IColumn[]>([])
+
+  useEffect(() => {
+    setColumns(tableColumns(slowQueryList || [], onColumnClick, orderBy, desc))
+    // eslint-disable-next-line
+  }, [slowQueryList])
+
+  useEffect(() => {
+    async function getSchemas() {
+      const res = await client.getInstance().statementsSchemasGet()
+      setSchemas(res?.data || [])
+    }
+    getSchemas()
+  }, [])
+
+  useEffect(() => {
+    async function getSlowQueryList() {
+      setLoading(true)
+      const res = await client
+        .getInstance()
+        .slowqueryListGet(
+          curSchemas.join(','),
+          desc,
+          limit,
+          curTimeRange?.end_time,
+          curTimeRange?.begin_time,
+          orderBy,
+          searchText
+        )
+      setLoading(false)
+      if (res?.data) {
+        setSlowQueryList(res.data || [])
+      }
+    }
+    getSlowQueryList()
+  }, [curTimeRange, curSchemas, orderBy, desc, searchText, limit, refreshTimes])
+
+  // TODO: refine
+  const location = useLocation()
+  useEffect(() => {
+    if (location.search === '?from=detail') {
```

Comment:
when will we need it?

Reply:
back from the detail page, we need to recover the previous search options, we can use Context as in the statements app, but it needs more code, so I want to try another simpler way.

Reply:
We can save options in the URL, as it also suffers from page refresh.

Reply:
Got what you mean, just want to try it, and should we do like this or store in session storage as you mentioned in #411

Reply:
I prefer to use session storage, so it also can work when switch apps, how do you think about it?